pkg/dataplane/inventory.go

```diff
@@ -39,51 +39,87 @@ import (
 // getAnsibleVarsFrom gets ansible vars from ConfigMap/Secret
 func getAnsibleVarsFrom(ctx context.Context, helper *helper.Helper, namespace string, ansible *dataplanev1.AnsibleOpts) (map[string]interface{}, error) {
 	result := make(map[string]interface{})
-	var value interface{}
 
 	for _, dataSource := range ansible.AnsibleVarsFrom {
 		configMap, secret, err := util.GetDataSourceCmSecret(ctx, helper, namespace, dataSource)
 		if err != nil {
 			return result, err
 		}
 
-		// AnsibleVars will override AnsibleVarsFrom variables.
-		// Process AnsibleVarsFrom first then allow AnsibleVars to replace existing values.
 		if configMap != nil {
-			for k, v := range configMap.Data {
-				if len(dataSource.Prefix) > 0 {
-					k = dataSource.Prefix + k
-				}
-
-				// Attempt to unmarshal the value into json. If that fails,
-				// assume it's a plain string.
-				err := json.Unmarshal([]byte(v), &value)
-				if err != nil {
-					value = v
-				}
-				result[k] = value
+			err := processConfigMapData(configMap.Data, dataSource.Prefix, result)
+			if err != nil {
+				return result, err
 			}
 		}
 
 		if secret != nil {
-			for k, v := range secret.Data {
-				if len(dataSource.Prefix) > 0 {
-					k = dataSource.Prefix + k
-				}
-				// Attempt to unmarshal the value into json. If that fails,
-				// assume it's a plain string.
-				err := json.Unmarshal(v, &value)
-				if err != nil {
-					value = string(v)
-				}
-				result[k] = value
+			err := processSecretData(secret.Data, dataSource.Prefix, result)
+			if err != nil {
+				return result, err
 			}
 		}
-
 	}
 	return result, nil
 }
 
+// processConfigMapData processes the key-value pairs from ConfigMap and adds them to the result map.
+func processConfigMapData(data map[string]string, prefix string, result map[string]any) error {
+	var value any
+	for k, v := range data {
+		if len(prefix) > 0 {
+			k = prefix + k
+		}
+
+		err := json.Unmarshal([]byte(v), &value)
```

Comment:
could use  https://pkg.go.dev/encoding/json#NewDecoder and then https://pkg.go.dev/encoding/json#Decoder.UseNumber to keep all numbers. and then could be smart about which ones to convert to int or float.

Reply:
Ok, I used the `decoder.UseNumber()`. This seems to work now:
```
[m3@osp-df-3 openstack-operator]$ oc get secret dataplanenodeset-openstack-edpm-bootc -o jsonpath='{.data.inventory}' | base64 -d | grep rhc_org
        rhc_organization: "1234567890"
```

Reply:
Prior to the change, it was in the inventory as:
```
        rhc_organization: 1.23456789e+09
```